### PR TITLE
Propose CONSTANT price identifier

### PR DESCRIPTION
**Pull request by smb2796**

### CODE REVIEW

File: UMIPs/umip-constant-priceid.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# TECHNICAL SPECIFICATIONS
+
+- Price Identifier Name: CONSTANT
+- Base Currency: NA
+- Quote currency: NA
+- Rounding: No rounding is necessary. The value returned should always be exactly equal to the value passed in ancillary data, or equal to 1 if no ancillary data is used.
+
+
+When converted from bytes to UTF-8, accompanying ancillary data should follow this format:
```
Comment (nicholaspai) [inline UMIPs/umip-constant-priceid.md:37]:
```suggestion
When converted from bytes to UTF-8, the ancillary data should be a dictionary containing a `constant` key like so:
```

File: UMIPs/umip-constant-priceid.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```
+constant:2
+```
+
+2 should be replaced with the constant value that the contract deployer wants to use.
```
Comment (nicholaspai) [inline UMIPs/umip-constant-priceid.md:42]:
```suggestion
The `constant` key should return the value that voters should resolve for this identifier. In this example, voters should vote 2.
```

File: UMIPs/umip-constant-priceid.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+When stored as bytes, the example above would be: 0x636f6e7374616e743a32.
```
Comment (nicholaspai) [inline UMIPs/umip-constant-priceid.md:44]: Can you provide a link or more details on converting the above into bytes?
```suggestion
When the simple example ancillary data dictionary "constant:2" is stored as bytes, the result would be: 0x636f6e7374616e743a32.
```

File: UMIPs/umip-constant-priceid.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# RATIONALE
+No rationale is needed. The motivation for this price identifier is explained in `motivation` and there is no plausible alternative for how to return a constant value.
+	
+# IMPLEMENTATION
+
+1. Look at the price request and get the ancillary data.
+2. Decode the ancillary data from bytes to UTF-8.
+3. If no ancillary data is provided, or there is ancillary data but it cannot be converted to the format in `Technical Specifications`, return 1. 
+4. If there is ancillary data that maps to the constant:value format, return the `constant` value.
```
Comment (nicholaspai) [inline UMIPs/umip-constant-priceid.md:54]:
```suggestion
4. If the ancillary data in UTF-8 contains a `constant` key, return the `constant` value.
```

File: UMIPs/umip-constant-priceid.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (smb2796) [reply UMIPs/umip-constant-priceid.md]: Yes, I think it should be. Will update

File: UMIPs/umip-constant-priceid.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+- Price Identifier Name: CONSTANT
+- Base Currency: NA
+- Quote currency: NA
+- Rounding: No rounding is necessary. The value returned should always be exactly equal to the value passed in ancillary data, or equal to 1 if no ancillary data is used.
+
+
+When converted from bytes to UTF-8, accompanying ancillary data should follow this format:
+```
+constant:2
```
[the remark this answers is not in the document]
Comment (smb2796) [reply UMIPs/umip-constant-priceid.md:39]: Agreed they absolutely should be!

File: UMIPs/umip-constant-priceid.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (smb2796) [reply UMIPs/umip-constant-priceid.md]: Actually I'm not sure. This number is purely for voter and bot usage, so would there be any hard requirement to scale this?

File: UMIPs/umip-constant-priceid.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+This constant identifier will also be the first price identifier to reference ancillary data.
+
+# Markets & Data Sources
+
+None. This price identifier is just a constant and does not refer to any market.
+
+# PRICE FEED IMPLEMENTATION
+
+This price identifier will use the [ExpressionPriceFeed](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/ExpressionPriceFeed.js). To use ancillary data with this price identifier, a features will need to be added to the price feeds to allow for the decoding and use of ancillary data. Since this price identifier can be used without ancillary data, it should not be blocked by the lack of ancillary data functionality in the price feeds.
```
Comment (pemulis) [inline UMIPs/umip-constant-priceid.md:27]: `a feature will need to be added`

**Review by mhairimca — APPROVED** [review]

File: UMIPs/umip-constant-priceid.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+# IMPLEMENTATION
+
+1. Query for the `ancillaryData` value from the price request.
+2. Decode the ancillary data from bytes to UTF-8.
+3. If no ancillary data is provided, or there is ancillary data but it cannot be converted to the format in `Technical Specifications`, return `1`. 
+4. If the ancillary data in UTF-8 contains a `constant` key, return the `constant` value. Voters should disregard any other key:value pairs included in the ancillary data.
+
+# SECURITY CONSIDERATIONS
+
+Adding this price identifier poses no risk to the UMA system. Since there should be no inconsistency in the result that should be returned, there is no risk of lack of determinism in this approach, as long as the price requester specifies their ancillary data correctly.
```
Comment (mhairimca) [inline UMIPs/umip-constant-priceid.md:58]: Looks good

**Pull request merged**